astropy/units/format/base.py

```diff
@@ -133,37 +122,49 @@ def to_string(
         ValueError
             If ``fraction`` is not recognized.
         """
+        string_components = cls._to_string_helper(unit, fraction)
+        if isinstance(string_components, str):
+            return string_components
+        if fraction is True or fraction == "inline":
+            return cls._format_inline_fraction(*string_components)
+        raise ValueError(
```

Comment:
That would work at runtime, but it wouldn't be very nice for type checkers.

Reply:
@mhvk, did you see my response?

Reply:
Ah, sorry, I did actually think about it, but now see that I forgot to write down my thoughts: I started grumpily, feeling that the typing tail was wagging the code code, but then realized that, really, it shows here that the API is perhaps wrong: why should `format="multiline"` ever lead to an error? Maybe a user prefers multiline if possible (say in a notebook), but is fine with a regular fraction if it is not. I.e,. shouldn't it just be a preference for how one wants the fraction formatted, with the meaning of fraction remaining whether to use a fraction at all (its truthiness), and its value giving a preference for how to do that? Let me ping @olebole, since he did much of the work here...

Obviously, with that logic, it does make much more sense have a separate method, or, for base, just `_format_multiline_fraction  = _format_inline_fraction`.

p.s. Note that if we do not go that route, we do need to adjust the docstring - it should be correct for the class on which it is defined. Which means it would need to be overridden elsewhere. Possibly another argument for going with always having two methods...

Reply:
@mhvk I am not sure here: in principle you seem to be right; however there are many places where "multiline" is semantically wrong; f.e. for fits or vounit formats. Using it there would likely indicate something not well-thought by the user, and raising an error instead of "best-effort" may be appropriate.
Also, we may have in future more options to fine-tune the unit representation, and probably "many" of them may be format specific (like spaces in FITS units, or specifics TeX style output); so I'd tend to keep raising the error in case of non-applicable options.

Reply:
Why do the unit formatters support both `fraction="inline"` and `fraction=False`? The `"multiline"` option is different enough that supporting it is justified (it allows some units to take up less horizontal space at the cost of more vertical space, which can be useful e.g. in plots or in tables), but are the `"inline"` and `False` options both really needed? Or is it too late to have that discussion?

Reply:
it is the difference between not using a fraction at all: `km / s` vs `km s^-1`, so definitely needed!

Reply:
@mhvk, are you happy with the design here or do you still think there is too much duplication between `Base.to_string()` and `Console.to_string()`?